### PR TITLE
unify sysconfig and command line directives

### LINKED ISSUE #976: num_eval_batches option is not in config.schema and does not seem to be read by armory
I'm running a config with `num_eval_batches` set to 1, but it just runs through the entire data set

I don't find that config option in the config.schema file

Is it not getting parsed?

Not sure if there's a simple baseline config I can provide as a minimal reproducible example that won't kick off training first (e.g. the `baseline_cifar10` config).
Just add `num_eval_batches` to any config with pre-trained models and see if it has an effect.

### PR DESCRIPTION
**Pull request by mwartell**
Fixes #976 

Cross-pollinate sysconfig and command-line overrides from config into arguments and arguments back into config so that output records actual run-time parameters.

WIP because it needs formal tests across the 9 enumerated cases.

### CODE REVIEW

**Comment by mwartell** [comment]
There are a few corner cases that cannot be fixed without making for really complicated semantics. If a scenario "sysconfig" contains, for example "skip_benign: true", there is no way to override that from the command-line. This happens because argparse marks unspecified booleans as False, not as None or otherwise undeclared. 

I had documented this quirk in the markdown, but thought more confusing than glossing over it. I might feel differently later.

**Comment by davidslater** [comment]
> I had documented this quirk in the markdown, but thought more confusing than glossing over it. I might feel differently later.

I think not documenting this is fine. I don't think this is something that we would often designate in the config and then want to undo at the command line. If that turns out to be a real use case, we can just expand the command line functionality.

**Review by davidslater — APPROVED** [review]
LGTM. One question, but doesn't require a change unless you think so.

I tested this with some of the non-standard command line invocations like `--gpus=all`, which also implicitly sets `--use-gpu`, and it seems to work fine.

File: armory/arguments.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    # u         u       nothing
+    # f         u       nothing
+    # d         u       args <- config
+    # u         f       nothing
+    # f         f       nothing
+    # d         f       args <- config
+    # u         d       config <- args
+    # f         d       config <- args
+    # d         d       config <- args
+
```
[the remark this answers is not in the document]
Comment (mwartell) [reply armory/arguments.py]: You are correct. This table accompanied code with a much higher cyclomatic complexity so I wanted to be sure I accounted for all cases. The undef cases were excluded already by the code you reviewed.

File: armory/arguments.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        if specified[name]:
+            new_spec[name] = specified[name]
+
+    # sysconfig gets updated with prioritized union
+    sysconfig.update(new_spec)
+
+    # new_args now gets the original namespace and all truthy members of the prioritized
+    # union
+    specified.update(new_spec)
+    new_args = argparse.Namespace(**specified)
```
[the remark this answers is not in the document]
Comment (mwartell) [reply armory/arguments.py]: When I was learning Python, the standard library was some of the best examples of Pythonic practice I had available. There's a lot of beautiful code in there (and a passel of accidents of history, too).

File: docs/configuration_files.md [hunk 3]
```diff
@@ -148,3 +148,32 @@ field of the configuration file under the subfield "defense_model," with the num
...
+```
+armory run scenario.json --num-eval-batches 100
+```
+Then the command line will override the sysconfig and 100 batches (not 5) will
+be run. In this example, `--skip-benign` will also be true because it is
+in the sysconfig block.
+
+No matter whether these attributes are specified on the command line, in sysconfig,
+or both, the output file will record the attributes as executed, so you have a
+record of how the evaluation ultimately ran.
```
[the remark this answers is not in the document]
Comment (mwartell) [reply docs/configuration_files.md:179]: I added a cross-reference link to `command_line.md` for the full enumeration. The `command_line.md` already links to here
